routing/routing_helpers.cpp

```diff
@@ -33,6 +33,16 @@ void ReconstructRoute(IDirectionsEngine * engine, IRoadGraph const & graph,
   if (engine)
     engine->Generate(graph, path, times, turnsDir, junctions, trafficSegs, cancellable);
 
+  if (cancellable.IsCancelled())
+    return;
+
+  // In case of any errors in IDirectionsEngine::Generate() |junctions| is empty.
+  if (junctions.empty())
```

Comment:
If it comes to crash in fabric (https://fabric.io/mapsme/android/apps/com.mapswithme.maps.pro/issues/5858f58b0aeb16625b352b6d)

The most likely routing was canceled while reconstructing path/generating turns. I check it two line above.

But as for your question, in BicycleDirectionsEngine::Generate() and PedestrianDirectionsEngine::Generate() may be:
1. If path.size() == 1 => junctions is empty;
2. If ReconstructPath() returns false => junctions is empty;